### PR TITLE
Update Partition range for new deployments

### PR DESCRIPTION
**Pull request by gregcorbett**
For new deployments, the `CloudRecord` database will now be partitioned by month for months that are +/- 18 months of Sept 2017 (now). A comment has been added to show how one might re-partition existing instances.

A 'NULL' partition has been added to hold any Records with an invalid `StartTime`. Hopefully this partition will be empty! But it will be useful if we ever have a Summariser that makes better use of the partitioning.

Partitioning has also been added to the travis CI test, to at least test the syntax of the file

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,6 +28,8 @@ before_script:
   # create an empty mysql database
   - mysql -u root -e "create database apel_rest"
   - mysql -u root apel_rest < schemas/10-cloud.sql
+  # partition the database, atleast to make sure the syntax is correct
```
Comment (tofu-rocketry) [inline .travis.yml]: "at least"

File: schemas/20-cloud-extra.sql [hunk 2]
```diff
@@ -3,47 +3,21 @@
 
 -- Partitioning for CloudRecords to aid query performance and monthly deletions
 
+-- More partitions can be added by "reorganizing" the pDefault parition. E.g.:
+/*
+ALTER TABLE JobRecords REORGANIZE PARTITION pDefault INTO (
```
Comment (tofu-rocketry) [inline schemas/20-cloud-extra.sql]: CloudRecords?

File: schemas/20-cloud-extra.sql [hunk 3]
```diff
@@ -3,47 +3,21 @@
...
+    PARTITION p2016_01 VALUES LESS THAN (TO_DAYS('2018-01-01')),
```
Comment (tofu-rocketry) [inline schemas/20-cloud-extra.sql]: Dates don't match partition names.

File: schemas/20-cloud-extra.sql [hunk 4]
```diff
@@ -3,47 +3,21 @@
...
-PARTITION p2015_06 VALUES LESS THAN (TO_DAYS('2015-07-01')),
-PARTITION p2015_07 VALUES LESS THAN (TO_DAYS('2015-08-01')),
-PARTITION p2015_08 VALUES LESS THAN (TO_DAYS('2015-09-01')),
-PARTITION p2015_09 VALUES LESS THAN (TO_DAYS('2015-10-01')),
-PARTITION p2015_10 VALUES LESS THAN (TO_DAYS('2015-11-01')),
-PARTITION p2015_11 VALUES LESS THAN (TO_DAYS('2015-12-01')),
-PARTITION p2015_12 VALUES LESS THAN (TO_DAYS('2016-01-01')),
-PARTITION p2016_01 VALUES LESS THAN (TO_DAYS('2016-02-01')),
-PARTITION p2016_02 VALUES LESS THAN (TO_DAYS('2016-03-01')),
+PARTITION pNull VALUES LESS THAN (0),
```
Comment (tofu-rocketry) [inline schemas/20-cloud-extra.sql]: How did you get the idea for this?

File: schemas/20-cloud-extra.sql [hunk 5]
```diff
@@ -3,47 +3,21 @@
...
+-- Partition pNull will contain Records with a NULL
```
Comment (tofu-rocketry) [inline schemas/20-cloud-extra.sql]: I can see how you might want to split this partition from the rest, but comments usually come before the line they refer to.

File: schemas/20-cloud-extra.sql [hunk 6]
```diff
@@ -53,17 +27,33 @@ PARTITION p2016_08 VALUES LESS THAN (TO_DAYS('2016-09-01')),
 PARTITION p2016_09 VALUES LESS THAN (TO_DAYS('2016-10-01')),
 PARTITION p2016_10 VALUES LESS THAN (TO_DAYS('2016-11-01')),
 PARTITION p2016_11 VALUES LESS THAN (TO_DAYS('2016-12-01')),
```
Comment (tofu-rocketry) [inline schemas/20-cloud-extra.sql]: Oops, no p2016_12 originally. Did this come from the grid version?

File: schemas/20-cloud-extra.sql [hunk 7]
```diff
@@ -53,17 +27,33 @@ PARTITION p2016_08 VALUES LESS THAN (TO_DAYS('2016-09-01')),
...
+PARTITION p2018_06 VALUES LESS THAN (TO_DAYS('2018-07-01')),
+PARTITION p2018_07 VALUES LESS THAN (TO_DAYS('2018-08-01')),
+PARTITION p2018_08 VALUES LESS THAN (TO_DAYS('2018-09-01')),
+PARTITION p2018_09 VALUES LESS THAN (TO_DAYS('2018-10-01')),
+PARTITION p2018_10 VALUES LESS THAN (TO_DAYS('2018-11-01')),
+PARTITION p2018_11 VALUES LESS THAN (TO_DAYS('2018-12-01')),
+PARTITION p2018_12 VALUES LESS THAN (TO_DAYS('2019-01-01')),
+PARTITION p2019_01 VALUES LESS THAN (TO_DAYS('2019-02-01')),
+PARTITION p2019_02 VALUES LESS THAN (TO_DAYS('2019-03-01')),
+PARTITION p2019_03 VALUES LESS THAN (TO_DAYS('2019-04-01')),
```
Comment (tofu-rocketry) [inline schemas/20-cloud-extra.sql]: We should try and automate this. It's a bit silly having to visually check these!

File: schemas/20-cloud-extra.sql (hunk 6 above)
Comment (gregcorbett) [reply schemas/20-cloud-extra.sql]: Doesn't look like it, I must have made the Oops when adding new values

File: schemas/20-cloud-extra.sql (hunk 7 above)
Comment (gregcorbett) [reply schemas/20-cloud-extra.sql]: +1

File: .travis.yml (hunk 1 above)
Comment (gregcorbett) [reply .travis.yml]: fixed

File: schemas/20-cloud-extra.sql (hunk 2 above)
Comment (gregcorbett) [reply schemas/20-cloud-extra.sql]: fixed

File: schemas/20-cloud-extra.sql (hunk 3 above)
Comment (gregcorbett) [reply schemas/20-cloud-extra.sql]: fixed

File: schemas/20-cloud-extra.sql (hunk 4 above)
Comment (gregcorbett) [reply schemas/20-cloud-extra.sql]: it was while looking into the Monthly Summariser and discovering that the first partition is always read. And [this bug report](https://bugs.mysql.com/bug.php?id=49754) confirms it as a feature.

File: schemas/20-cloud-extra.sql (hunk 5 above)
Comment (gregcorbett) [reply schemas/20-cloud-extra.sql]: changed

**Comment by gregcorbett** [comment]
changes made

**Comment by gregcorbett** [comment]
I had to rebase to bring the branch up-to-date. Can you re-review it please?

**Pull request merged**